### PR TITLE
RCT-150: 30 Days of React - Day 1

### PR DESCRIPTION
**Pull request by jhrasco-ct**
## Related Tickets

- [RCT-150: React Training](https://codeandtheory.atlassian.net/browse/RCT-150)

## Description

Completed the exercises in the [30 Days of React - Day 1](https://github.com/jhrasco-ct/30-Days-Of-React/blob/master/01_Day_JavaScript_Refresher/01_javascript_refresher.md).

## Validation

1. Open each JavaScript file containing the code for each exercise.
2. Verify that the solutions written are correct and satisfy the challenges.
3. To verify, open the **index.html** file and inspect the output using the browser's console.

### CODE REVIEW

File: solutions/day-01/exercise1/level1/level1.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+const emptyArray = [];
+
+// 2. Declare an array with more than 5 number of elements
+const array = [1, 2, 3, 4, 5, 6, 7];
+
+// 3. Find the length of your array
+const length = array.length;
+
+// 4. Get the first item, the middle item and the last item of the array
+const firstItem = array[0];
```
Comment (ianabalus) [inline solutions/day-01/exercise1/level1/level1.js:11]: The code is ok, but just want to note that we can consider using destructuring assignments.
```
const [firstItem] = array;
```

File: solutions/day-01/exercise1/level2/main.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+} else {
+  webTechs.push("Sass");
+  console.log(webTechs);
+}
+
+// 6. Concatenate the following two variables and store it in a fullStack variable.
+const frontEnd = ["HTML", "CSS", "JS", "React", "Redux"];
+const backEnd = ["Node", "Express", "MongoDB"];
+
+const fullStack = frontEnd.concat(backEnd);
```
Comment (ianabalus) [inline solutions/day-01/exercise1/level2/main.js:62]: Another way to merge two variables is by using the spread operator
```
const fullStack = [...frontEnd, ...backEnd]
```

File: solutions/day-01/exercise3/level2/level2.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    skills.includes("node")
+  ) {
+    mernDevelopers.push(user);
+  }
+}
+
+console.log(mernDevelopers);
+
+// 4. Set your name in the users object without modifying the original users object
+const usersCopy = Object.assign({}, users);
```
Comment (ianabalus) [inline solutions/day-01/exercise3/level2/level2.js:52]: Another way to clone an object is by using the spread operator.
```
const usersCopy = {...users};
```
Spread operator is handy when dealing with arrays and objects.

File: solutions/day-01/exercise3/level3/level3.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+// 3.2 Create a function called averageRating which calculate the average rating of a product
+const averageRating = (productId) => {
+  const product = products.find((product) => product._id === productId);
+  if (product) {
+    const ratings = product.ratings.map((rating) => rating.rate);
+    const total = ratings.reduce((total, rating) => total + rating, 0);
+    return total / ratings.length;
+  } else {
+    return 0;
+  }
```
Comment (ianabalus) [inline solutions/day-01/exercise3/level3/level3.js:176]: If a function only has two return possibilities, there is no need to use the `else` block around the non-conditional return. Make it as a default return instead.
```
if (product) {
// statements
const total = ratings.reduce...
return total
}

return 0;
```

**Review by ianabalus — APPROVED** [review]
I will approve this PR, but I also left some feedback for you to review.

**Comment by ianabalus** [comment]
@jhrasco-ct, please give me write access so I can merge your PRs. Thank you!

**Comment by jhrasco-ct** [comment]
> @jhrasco-ct, please give me write access so I can merge your PRs. Thank you!

@ianabalus, I resent you the invite to be a collaborator.

Also, thanks for your helpful PR feedback.

**Pull request merged**